crux-lucene/src/crux/lucene.clj

```diff
@@ -0,0 +1,187 @@
+(ns crux.lucene
+  (:require [clojure.spec.alpha :as s]
+            [crux.codec :as cc]
+            [crux.db :as db]
+            [crux.io :as cio]
+            [crux.memory :as mem]
+            [crux.query :as q]
+            [crux.system :as sys]
+            [clojure.tools.logging :as log])
+  (:import crux.codec.EntityTx
+           java.io.Closeable
+           java.nio.file.Path
+           org.apache.lucene.analysis.Analyzer
+           org.apache.lucene.analysis.standard.StandardAnalyzer
+           [org.apache.lucene.document Document Field Field$Store StringField TextField]
+           [org.apache.lucene.index DirectoryReader IndexWriter IndexWriterConfig Term]
+           org.apache.lucene.queries.function.FunctionScoreQuery
+           org.apache.lucene.queryparser.classic.QueryParser
+           [org.apache.lucene.search BooleanClause$Occur BooleanQuery$Builder DoubleValuesSource IndexSearcher Query ScoreDoc TermQuery]
+           [org.apache.lucene.store Directory FSDirectory]))
+
+(def ^:dynamic *node*)
+
+(defrecord LuceneNode [directory analyzer]
+  java.io.Closeable
+  (close [this]
+    (doseq [^Closeable c [directory]]
+      (.close c))))
+
+(defn- id->stored-bytes [eid]
+  (mem/->on-heap (cc/->value-buffer eid)))
+
+(defn- ^String eid->str [eid]
+  (mem/buffer->hex (cc/->id-buffer eid)))
```

Comment:
think `(str (cc/new-id eid))` does the same thing here without low-level buffer stuff